dmoj/problem.py

```diff
@@ -188,27 +170,50 @@ def _resolve_archive_files(self):
 
 
 class ProblemDataManager(dict):
-    def __init__(self, problem, **kwargs):
+    def __init__(self, problem_root_dir, **kwargs):
         super().__init__(**kwargs)
-        self.problem = problem
+        self.problem_root_dir = problem_root_dir
         self.archive = None
 
     def __missing__(self, key):
         try:
-            with open(os.path.join(self.problem.root_dir, key), 'rb') as f:
+            with open(os.path.join(self.problem_root_dir, key), 'rb') as f:
                 return f.read()
         except IOError:
             if self.archive:
                 zipinfo = self.archive.getinfo(key)
                 with self.archive.open(zipinfo) as f:
                     return f.read()
-            raise KeyError('file "%s" could not be found in "%s"' % (key, self.problem.root_dir))
+            raise KeyError('file "%s" could not be found in "%s"' % (key, self.problem_root_dir))
 
     def __del__(self):
         if self.archive:
             self.archive.close()
 
 
+class ProblemConfig(ConfigNode):
+    def __init__(self, problem_data, meta={}):
+        try:
+            doc = yaml.safe_load(problem_data['init.yml'])
+            if not doc:
+                raise InvalidInitException('I find your lack of content disturbing.')
+            super().__init__(
+                doc,
+                defaults={
+                    'wall_time_factor': 3,
+                    'output_prefix_length': 0 if 'signature_grader' in doc else 64,
+                    'output_limit_length': 25165824,
+                    'binary_data': False,
+                    'short_circuit': True,
+                    'points': 1,
+                    'symlinks': {},
+                    'meta': meta,
+                },
+            )
+        except (IOError, KeyError, ParserError, ScannerError) as e:
```

Comment:
Should this be scoped just around the `safe_load`, or can the `super().__init__` actually raise?